### PR TITLE
Add applet to transfer CMV12K sensor data to DRAM

### PR DESCRIPTION
**Pull request by tpwrules**
Adds an applet to receive test data from the CMV12K sensor and store it in the main DRAM. Remapping is not implemented yet.

To do:

- [x] Fix `DramPacketRingbufferStreamWriter` (PR #24 broke it)
- [ ] Fix the fact that received image is 8192x1536 instead of correct 4096x3072
- [ ] Possibly validate pattern after transfer
- [ ] Possibly add HDMI output

### CODE REVIEW

**Review by rroohhh — CHANGES_REQUESTED** [review]
Just some preliminary notes, Ill take a deeper look later

File: naps/cores/dram_packet_ringbuffer/cpu_if.py [hunk 1]
```diff
@@ -53,12 +53,9 @@ def elaborate(self, platform):
...
-            sleep(0.01)
-
-        self.current_read_buffer = self.current_read_buffer % self.num_buffers
-        os.system("dd if=/dev/mem bs=4096 skip={} count={} iflag=skip_bytes,count_bytes of='{}'"
-                  .format(self.current_buffer_base, self.current_buffer_length, filename))
 
+        buf = (self.current_write_buffer - 1) % self.num_buffers
+        base = getattr(self, f"buffer{buf}_base")
+        length = getattr(self, f"buffer{buf}_level")
+        os.system(f"dd if=/dev/mem bs=4096 skip={base} count={length} iflag=skip_bytes,count_bytes of='{filename}'")
```
[the remark this answers is not in the document]
Comment (tpwrules) [reply naps/cores/dram_packet_ringbuffer/cpu_if.py:61]: I think that would be nice too. Should we use `DevMemAccessor`, or just mmap /dev/mem directly? The downside would be that `DevMemAccessor` can only read one word at a time.

File: naps/cores/dram_packet_ringbuffer/cpu_if.py [hunk 2]
```diff
@@ -33,17 +33,17 @@ def __init__(self, writer: DramPacketRingbufferStreamWriter):
 
         m = Module()
 
-        self.current_write_buffer = StatusSignal(range(writer.n_buffers))
-        m.d.comb += self.current_write_buffer.eq(writer.current_write_buffer)
+        self.num_buffers = writer.n_buffers
+        self.current_write_buffer = StatusSignal(self.num_buffers)
+        m.d.sync += self.current_write_buffer.eq(writer.current_write_buffer)
```
[the remark this answers is not in the document]
Comment (tpwrules) [reply naps/cores/dram_packet_ringbuffer/cpu_if.py:38]: To improve timing. It should not affect correctness I don't think.

File: naps/cores/dram_packet_ringbuffer/cpu_if.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (rroohhh) [reply naps/cores/dram_packet_ringbuffer/cpu_if.py:61]: I think just mmap /dev/mem directly first, there are some changes to `DevMemAccessor` in the pipe, that will allow to read a arbitrary number of words at once, so we can switch to that when its ready.

**Review by rroohhh — APPROVED** [review]
Nice work!

**Pull request merged**